app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php

```diff
@@ -203,7 +203,8 @@ public function execute()
                     $this->messageManager->addErrorMessage(
                         __(
                             'Attribute code "%1" is invalid. Please use only letters (a-z or A-Z), ' .
-                            'numbers (0-9) or underscore(_) in this field, first character should be a letter.',
+                            'numbers (0-9) or underscore(_) in this field, first character should be a letter, ' .
+                            'and length should be less than 30 characters.',
```

Comment:
Please make a separate check for length which appears earlier than this.

Also, 30 chars seems to be allowed from regexp.

Reply:
@orlangur sir
sure i am adding it.

Reply:
@orlangur sir ,
i have added seperate length check before string validation.  
```
 $validatorAttrCode = new \Zend_Validate_Regex(
                    ['pattern' => '/^[a-zA-Z\x{600}-\x{6FF}][a-zA-Z\x{600}-\x{6FF}_0-9]{0,30}$/u']
                );
```

Reply:
@orlangur 
sir , please review the changes

Reply:
@orlangur 
sir, please review the changes.
Thank you.